packages/core/admin/admin/src/content-manager/pages/ListView/index.js

```diff
@@ -483,6 +429,19 @@ function ListView({
     }
   };
 
+  // Refetch the data and update the main data without loading
+  const refreshDataOnBackground = async () => {
+    setIsLoadingOnBackground(true);
+    const endpoint = getRequestUrl(`collection-types/${slug}${params}`);
+
+    const {
+      data: { results, pagination: paginationResult },
+    } = await fetchClient.get(endpoint);
+
+    getDataSucceeded(paginationResult, results);
+    setIsLoadingOnBackground(false);
+  };
```

Comment:
we should be using react-query for this sort of thing, not a custom function.

Reply:
Maybe you don't need to do a big refactor for this. What about handling this with a useQuery that is enabled based on a certain condition: https://tanstack.com/query/v3/docs/react/guides/disabling-queries. Could that work?

Reply:
Otherwise, I agree the refactor should be another PR as discussed here: https://github.com/strapi/strapi/pull/17074#discussion_r1246678871, and I would be happy to work on it :)